### PR TITLE
Incorporate action result executors and a set of helper methods around that

### PR DESCRIPTION
**Pull request by filipw**
@khellang this is based on your excellent blog post and modelled after the helper methods on `ControllerBase`. I did not port the ones that didn't make sense (like `*ToAction`) or the ones that were simply convenience calls into `HttpContext` (like `Challenge`, `SignIn` and so on).

Since there are plenty of action results that do not have a corresponding executor (most of the simple ones like `OkResult` or `OkObjectResult`) those have been adapted accordingly to use `HttpResponse` directly or other usable results like `ObjectResult` with modified properties.

### CODE REVIEW

File: src/WebApiContrib.Core/Results/HttpContextExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+        public static Task WriteActionResult<TResult>(this HttpContext context, TResult result) where TResult : IActionResult
+        {
+            var executor = context.RequestServices.GetService<IActionResultExecutor<TResult>>();
+
+            if (executor == null)
+            {
+                throw new InvalidOperationException($"No action result executor for {typeof(TResult).FullName} registered.");
+            }
+
+            var routeData = context.GetRouteData() ?? new RouteData();
```
Comment (khellang) [inline src/WebApiContrib.Core/Results/HttpContextExtensions.cs]: Should we cache these empty values?

Comment (filipw) [reply src/WebApiContrib.Core/Results/HttpContextExtensions.cs]: I thought about that, but technically they are mutable so I guess something bad might happen

Comment (khellang) [reply src/WebApiContrib.Core/Results/HttpContextExtensions.cs]: Hmm. That's a good point 👍

File: src/WebApiContrib.Core/Results/HttpContextExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+using System.IO;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace WebApiContrib.Core.Results
+{
+    public static class HttpContextExtensions
+    {
+        // adapted from https://kristian.hellang.com/using-mvc-result-executors-in-middleware/
```
Comment (khellang) [inline src/WebApiContrib.Core/Results/HttpContextExtensions.cs]: Thanks 😘

Comment (filipw) [reply src/WebApiContrib.Core/Results/HttpContextExtensions.cs]: I forgot earlier 😞

**Pull request merged**